src/containers/Topbar/Topbar.js

```diff
@@ -85,12 +93,48 @@ class Topbar extends React.PureComponent {
     this.handleCancel = this.handleCancel.bind(this);
     this.onAddressInputChange = this.onAddressInputChange.bind(this);
     this.onAddressDropdownClick = this.onAddressDropdownClick.bind(this);
-    this.getSelectedAddressObject = this.getSelectedAddressObject.bind(this);
     this.onCopyClicked = this.onCopyClicked.bind(this);
   }
 
   componentWillMount() {
-    this.props.listUnspent();
+    const { onGetBlockCount, listUnspent } = this.props;
+
+    (function startPoll() {
+      onGetBlockCount();
+      listUnspent();
+      setTimeout(startPoll, POOL_INTERVAL);
+    }());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {
+      onGetBlockCount, getBotBalance, selectedWalletAddress,
+    } = this.props;
+
+
+    // Call API to retrieve BOT balance if wallet addresses have changed
+    if (!_.isEqual(this.props.walletAddrs, nextProps.walletAddrs)) {
+      _.each(nextProps.walletAddrs, (addressObj) => {
+        const ownerAddress = addressObj.address;
+        let senderAddress;
+
+        // Determine an address that has enough qtum for gas to be a sender
+        if (addressObj.qtum >= MINIMAL_GAS_FEE) {
+          senderAddress = addressObj.address;
+        } else {
+          const objWithQtum = _.find(nextProps.walletAddrs, (item) => item.qtum >= MINIMAL_GAS_FEE);
+
+          if (objWithQtum) {
+            senderAddress = objWithQtum.address;
+          }
+        }
+
+        // Found any address with balance greater than minimal gas fee
+        if (senderAddress) {
+          getBotBalance(ownerAddress, senderAddress);
```

Comment:
do we want to get the botBalance of every address? or just the one that is selected?

Reply:
We plan to show balance of every address in his wallet, that way he can see all balances in topbar dropdown.